packages/typespec-azure-resource-manager/lib/Legacy/operations.tsp

```diff
@@ -0,0 +1,272 @@
+     */
+    #suppress "@azure-tools/typespec-azure-core/no-private-usage"
+    @autoRoute
+    @doc("Create a {name}", Resource)
+    @armResourceCreateOrUpdate(Resource)
+    @put
+    CreateOrUpdateSync<
+      Resource extends Azure.ResourceManager.CommonTypes.Resource,
+      Parameters extends {} = {},
+      Response extends {} = ArmResourceUpdatedResponse<Resource> | ArmResourceCreatedSyncResponse<Resource>
+    >(
+      ...ParentParameters,
+      ...ResourceTypeParameter,
+      ...Parameters,
+      @doc("Resource create parameters.") @bodyRoot resource: Resource,
+    ): Response | ErrorType;
+
+    /**
+     * @dev A long-running resource Update (PATCH)
+     * @template Resource the resource being created or updated
+     * @template PatchModel the PATCH request model
+     * @template LroHeaders Optional.  Allows overriding the lro headers returned on resource create
+     * @template Parameters Optional. Additional parameters after the path parameters
+     * @template Response Optional. The success response(s) for the PATCH operation
+     */
+    @autoRoute
+    @doc("Update a {name}", Resource)
+    @armResourceUpdate(Resource)
+    @patch
+    CustomPatchAsync<
+      Resource extends Azure.ResourceManager.CommonTypes.Resource,
+      PatchModel extends {} = Azure.ResourceManager.Foundations.TagsUpdateModel<Resource>,
+      LroHeaders extends TypeSpec.Reflection.Model = ArmLroLocationHeader<
+        Azure.Core.StatusMonitorPollingOptions<ArmOperationStatus>,
+        Resource,
+        string
+      > &
+        Azure.Core.Foundations.RetryAfterHeader,
+      Parameters extends {} = {},
+      Response extends {} = ArmResponse<Resource> | ArmAcceptedLroResponse<
+        "Resource update request accepted.",
+        LroHeaders
+      >
+    >(
+      ...ParentParameters,
+      ...ResourceTypeParameter,
+      ...Parameters,
+      @doc("The resource properties to be updated.") @bodyRoot properties: PatchModel,
+    ): Response | ErrorType;
+
+    /**
+     * @dev A synchronous resource Update (PATCH)
+     * @template Resource the resource being created or updated
+     * @template PatchModel the PATCH request model
+     * @template Parameters Optional. Additional parameters after the path parameters
+     * @template Response Optional. The success response(s) for the PATCH operation
+     */
+    @autoRoute
+    @doc("Update a {name}", Resource)
+    @armResourceUpdate(Resource)
+    @patch
+    CustomPatchSync<
+      Resource extends Azure.ResourceManager.CommonTypes.Resource,
+      PatchModel extends {} = Azure.ResourceManager.Foundations.TagsUpdateModel<Resource>,
+      Parameters extends {} = {},
+      Response extends {} = ArmResponse<Resource>
+    >(
+      ...ParentParameters,
+      ...ResourceTypeParameter,
+      ...Parameters,
+      @doc("The resource properties to be updated.") @bodyRoot properties: PatchModel,
+    ): Response | ErrorType;
+
+    /**
+     * @dev Delete a resource asynchronously
```

Comment:
I'd like us to be explicit that these APIs are subject to change, and should only be used in converting legacy APIs to typespec.

If we could, I would consider an experimental types export like this: https://github.com/microsoft/typespec/blob/main/packages/events/package.json#L28

But we would have to validate that this works with imports and standard tsp compilation.  The best current solution might be docs and either a decorator or rule that marks these types as volatile, and issues a warning if you use them (perhaps through conbining a decorator and linting rule, so that it could be globally suppressed in config).